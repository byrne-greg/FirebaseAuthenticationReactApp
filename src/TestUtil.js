import React from "react";
import { MemoryRouter } from "react-router-dom";

const withTestRouter = Component => <MemoryRouter>{Component}</MemoryRouter>;

export { withTestRouter };
