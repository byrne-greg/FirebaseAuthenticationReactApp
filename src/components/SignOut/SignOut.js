import React, { useEffect, useContext } from "react";
import { withRouter } from "react-router-dom";
import { FirebaseContext } from "../Firebase";
import * as ROUTES from "../../constants/routes";
import messages from "./messages";

const SignOut = ({ history }) => {
  useEffect(() => {
    setTimeout(() => history.push(ROUTES.LANDING), 5000);
  }, [history]);
  return <h1>{messages.successfulSignOut}</h1>;
};

const SignOutButtonComponent = ({ history }) => {
  const firebase = useContext(FirebaseContext);

  return (
    <button
      id="sign-out-button"
      type="button"
      onClick={() => {
        firebase.logout();
        history.push(ROUTES.SIGN_OUT);
      }}
    >
      {messages.signOut}
    </button>
  );
};
const SignOutButton = withRouter(SignOutButtonComponent);

export default withRouter(SignOut);
export { SignOutButton };
