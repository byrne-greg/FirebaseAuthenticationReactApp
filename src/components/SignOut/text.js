export default {
  successfulSignOut: "You have successfully signed out",
  signOut: "Sign Out"
};
