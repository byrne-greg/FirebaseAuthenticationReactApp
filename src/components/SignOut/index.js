import SignOut, { SignOutButton } from "./SignOut";

export default SignOut;
export { SignOutButton };
