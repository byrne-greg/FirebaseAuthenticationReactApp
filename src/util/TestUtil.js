import React from "react";
import { MemoryRouter } from "react-router-dom";
import { FirebaseContext } from "../components/Firebase";
import { AuthUserContext } from "../components/Session";

const withMockFirebase = (mockFirebase, Component) => (
  <FirebaseContext.Provider value={mockFirebase}>
    {Component}
  </FirebaseContext.Provider>
);

const withMockAuthUserContext = (value, Component) => (
  <AuthUserContext.Provider value={value}>{Component}</AuthUserContext.Provider>
);

const withTestRouter = Component => <MemoryRouter>{Component}</MemoryRouter>;

export { withTestRouter, withMockFirebase, withMockAuthUserContext };
