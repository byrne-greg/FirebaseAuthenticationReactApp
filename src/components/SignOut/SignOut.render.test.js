import React from "react";
import { withTestRouter } from "../../TestUtil";
import { render, cleanup } from "@testing-library/react";
import SignOut, { SignOutButton } from ".";

afterEach(cleanup);

describe("SignOut render tests", () => {
  it("page renders as expected", () => {
    // GIVEN

    // WHEN
    const page = render(withTestRouter(<SignOut />));

    // THEN
    expect(page.baseElement).toMatchSnapshot();
  });

  it("button renders as expected", () => {
    // GIVEN

    // WHEN
    const button = render(withTestRouter(<SignOutButton />));

    // THEN
    expect(button.baseElement).toMatchSnapshot();
  });
});
