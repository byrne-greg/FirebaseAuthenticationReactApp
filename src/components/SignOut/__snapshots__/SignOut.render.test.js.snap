// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignOut render tests button renders as expected 1`] = `
<body>
  <div>
    <button
      id="sign-out-button"
      type="button"
    >
      Sign Out
    </button>
  </div>
</body>
`;

exports[`SignOut render tests page renders as expected 1`] = `
<body>
  <div>
    <h1>
      You have successfully signed out
    </h1>
  </div>
</body>
`;
